modelql-core/src/commonMain/kotlin/org/modelix/modelql/core/MappingStep.kt

```diff
@@ -36,6 +36,10 @@ class MappingStep<In, Out>(val query: MonoUnboundQuery<In, Out>) : MonoTransform
         return query.requiresWriteAccess()
     }
 
+    override fun needsCoroutineScope(): Boolean {
```

Comment:
Why is this override needed?
What if the `MappingStep.requiresWriteAccess` is `true`?

(It can also be simplified to `override fun needsCoroutineScope() = false`.)

Reply:
Hm, ok. I still do not fully understand it.

Anyway, now we have many other `MonoTransformingStep`s like `MapIfNotNullStep` that break with the current change because they are not adapted like `MappingStep` here.

Reply:
Also `FluxTransformingStep`s or `TransformingStep`s might be affected. For now I only tested, that `MonoTransformingSteps` is broken.

Reply:
Why don't we have tests that uncover those problems?

Reply:
Also using `.map` in the provided test fails:

```
@Test
    fun `chained write operations are executed once`() = runTypedModelQLTest { client ->
       ...

        client.query { root ->
            root.children("classes").ofConcept(C_ClassConcept)
                .first()
                .map {
                    it.addToMember(C_StaticMethodDeclaration)
                }
                .setName(newChildName.asMono())
        }

        ...
        assertEquals(newChildName, actualChildren.last().name)
    }
  ```

Reply:
> Why don't we have tests that uncover those problems?

I guess this is a fragile base class problem.

`ModelQLClientTest.testCaching()` tested correct implementation in `MonoTransformingSteps/TransformingStep` using  `MappingStep` as an example. But with changes to the base class things broke 🤷

Reply:
@mhuster23 Do you think it makes still sense to fix it with the Flow-Implementation?
The new Bulk-Sync in [MODELIX-914](https://issues.modelix.org/issue/MODELIX-914) will possibly get rid of this bug anyway.